dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -49,8 +49,32 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 {
     winrt::Microsoft::Windows::AppNotifications::AppNotificationManager AppNotificationManager::Default()
     {
-        static auto appNotificationManager{winrt::make<AppNotificationManager>()};
-        return appNotificationManager;
+        auto appProperties{ winrt::CoreApplication::Properties() };
+        // Store the PushNotificationManager in the COM static store
+        auto storedAppNotificationManager{ appProperties.TryLookup(STORED_APPNOTIFICATION_MANAGER_KEY) };
```

Comment:
No lock :(

Reply:
And you have read-write races too because of the absence of a lock around this whole operation.